packages/protocol/database/manager.go

```diff
@@ -95,16 +228,22 @@ func (m *Manager) Shutdown() {
 //   index 1 -> db 0
 //   index 2 -> db 2
 func (m *Manager) getDBInstance(index epoch.Index) (db *dbInstance) {
-	startingIndex := index / epoch.Index(m.optsGranularity) * epoch.Index(m.optsGranularity)
+	startingIndex := m.computeDBBaseIndex(index)
 	db, exists := m.dbs.Get(startingIndex)
 	if !exists {
 		db = m.createDBInstance(startingIndex)
 		m.dbs.Set(startingIndex, db)
 	}
+	db.lastAccessed = time.Now()
 
 	return db
 }
 
+func (m *Manager) computeDBBaseIndex(index epoch.Index) epoch.Index {
```

Comment:
Should epoch.Index be used in the context of DB index? It could be slightly confusing

Reply:
Since it's an alias for `int64` maybe just create a new alias e.g. `DBBaseIndex`.

Reply:
I tried to use it with this type but turned out to be awkward because lots of type conversions and we still need to handle `epoch.Index` in many places. Also the DBBaseIndex is technically only `index / epoch.Index(m.optsGranularity) * epoch.Index(m.optsGranularity)`. So for now I'll leave it with `epoch.Index`.